.github/workflows/test_hipcub.yml

```diff
@@ -69,4 +69,5 @@ jobs:
             --test-dir ${THEROCK_BIN_DIR}/hipcub \
             --output-on-failure \
             --parallel 8 \
-            --timeout 360
+            --timeout 360 \
+            --rerun-failed
```

Comment:
https://cmake.org/cmake/help/latest/manual/ctest.1.html#cmdoption-ctest-rerun-failed

Looks like you may want to use something like `--repeat until-pass:3` instead of `--rerun-failed`.